### ISSUE TITLE
interpolateAtSample bug?

### ISSUE THREAD

**Issue by elect86** [g-truc/ogl-samples#68]
I can't run gl-400-fbo-multisample and all the others who uses the `interpolateAtSample`

```
in vert
{
    vec2 Texcoord;
    //sample vec2 Texcoord;
} Vert;

layout(location = FRAG_COLOR, index = 0) out vec4 Color;

void main()
{
    Color = texture(Diffuse, interpolateAtSample(Vert.Texcoord, gl_SampleID));
    //Color = texture(Diffuse, Vert.Texcoord);
}
```

Asking on the irc opengl, a guy suggested me to use the commented out lines instead, and that works. He also said it is probabily a glsl compiler bug... what do you think?

Edit: https://devtalk.nvidia.com/default/topic/914874/opengl/glsl-compiler-bug-on-interpolateatsample-/

**Comment by pixeljetstream** [comment]
There is a known compiler bug for this, if you remove "Vert" at the end of the "in" definition and use Texcoord directly it should work in the interpolateAtSample function.

**Comment by Groovounet** [comment]
Hi @elect86,

What drivers / GPU are you using? Do you see an GLSL compiler error? What is it?
On a GTX 970, it works just fine for me. (tm)

Thanks,
Christophe

**Comment by Groovounet** [comment]
This issue is fixed in 4.5.4 branch. 

Thanks for reporting,
Christophe
